src/Storage/Device/MinIO.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+namespace Utopia\Storage\Device;
+
+use Utopia\Storage\Device\S3;
+
+class MinIO extends S3
+{
+    /**
+     * Regions constants
+     *
+     */
+    const EU_CENTRAL_1='eu-central-1';
+    const US_SOUTHEAST_1='us-southeast-1';
+    const US_EAST_1='us-east-1';
+    const AP_SOUTH_1='ap-south-1';
+
+    /**
+     * Object Storage Constructor
+     *
+     * @param string $root
+     * @param string $accessKey
+     * @param string $secretKey
+     * @param string $protocol
+     * @param string $host
+     * @param string $bucket
+     * @param string $region
+     * @param string $acl
+     */
+    public function __construct(string $root, string $accessKey, string $secretKey, string $protocol, string $host, string $bucket, string $region = self::EU_CENTRAL_1, string $acl = self::ACL_PRIVATE)
+    {
+        parent::__construct($root, $accessKey, $secretKey, $bucket, $region, $acl);
+        
+        $this->protocol = $protocol;
+        $this->headers['host'] = $host;
+    }
+
+    /**
+     * Get list of objects in the given path.
+     *
+     * @param string $path
+     * 
+     * @throws \Exception
+     *
+     * @return array
+     */
+    public function listObjects($prefix = '', $maxKeys = 1000, $continuationToken = '')
```

Comment:
```suggestion
    private function listObjects($prefix = '', $maxKeys = 1000, $continuationToken = '')
```

Please change S3 listObjects() access  modifiers to protected in order for you to override it.